File: app/src/test/java/com/github/colinjeremie/willyoufindit/activities/PickACategoryActivityTest.java
package com.github.colinjeremie.willyoufindit.activities;


import android.os.Build;

import com.github.colinjeremie.willyoufindit.BuildConfig;
import com.github.colinjeremie.willyoufindit.R;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricGradleTestRunner;
import org.robolectric.annotation.Config;
import org.robolectric.internal.ShadowExtractor;
import org.robolectric.shadows.ShadowActivity;

@RunWith(RobolectricGradleTestRunner.class)
@Config(constants = BuildConfig.class, sdk = Build.VERSION_CODES.LOLLIPOP)
public class PickACategoryActivityTest {

    private ShadowActivity mActivity;

    @Before
    public void setUp(){
        mActivity = (ShadowActivity) ShadowExtractor.extract(Robolectric.buildActivity(PickACategoryActivity.class).create().resume().get());
    }

    @Test
    public void shouldDisplayRadioFragmentOnClick() throws Exception{
        mActivity.findViewById(R.id.choose_category_radio_btn).performClick();
        Assert.assertEquals(RadioActivity.class.getName(), mActivity.getNextStartedActivity().getComponent().getClassName());
    }

    @Test
    public void shouldDisplayGenreFragmentOnClick() throws Exception{
        mActivity.findViewById(R.id.choose_category_genre_btn).performClick();
        Assert.assertEquals(GenreActivity.class.getName(), mActivity.getNextStartedActivity().getComponent().getClassName());
    }
}

File: app/src/test/java/com/github/colinjeremie/willyoufindit/activities/PickACategoryActivityTest.kt
package com.github.colinjeremie.willyoufindit.activities


import android.os.Build
import com.github.colinjeremie.willyoufindit.BuildConfig
import com.github.colinjeremie.willyoufindit.R
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config
import org.robolectric.shadow.api.Shadow
import org.robolectric.shadows.ShadowActivity

@RunWith(RobolectricTestRunner::class)
@Config(constants = BuildConfig::class, sdk = [(Build.VERSION_CODES.LOLLIPOP)])
class PickACategoryActivityTest {

    @Test
    @Throws(Exception::class)
    fun should_display_radio_fragment_on_click() {
        // Given
        val activity = Shadow.extract<ShadowActivity>(Robolectric.buildActivity(PickACategoryActivity::class.java).create().resume().get())

        // When
        activity.findViewById(R.id.choose_category_radio_btn).performClick()

        // Then
        Assert.assertEquals(RadioActivity::class.java.name, activity.nextStartedActivity.component!!.className)
    }

    @Test
    @Throws(Exception::class)
    fun should_display_genre_fragment_on_click() {
        // Given
        val activity = Shadow.extract<ShadowActivity>(Robolectric.buildActivity(PickACategoryActivity::class.java).create().resume().get())

        // When
        activity.findViewById(R.id.choose_category_genre_btn).performClick()

        // Then
        Assert.assertEquals(GenreActivity::class.java.name, activity.nextStartedActivity.component!!.className)
    }
}
